### DISCUSSION TITLE
Poor man's "combined row" across multiple monitors

### DISCUSSION THREAD

**Discussion by meghprkh** [YaLTeR/niri#810] [Ideas]
I have been using two small Full HD monitors for some time side to side. The current workflow I understand / usage is more optimized for bigger / wider displays.

Kinda like  (image from internet search)
![image](https://github.com/user-attachments/assets/46e228c1-4c50-44b5-9c49-5c2d7d79a33f)

**Is there a way to make these outputs "virtual" and combine them as one "row / swimlane"?**

This probably requires some change in the behaviour of where "columns edges" are placed.

I will try to explain "intuitive behaviour according to me":

- Notation:
  - Monitor boundary - `|`
  - "Combined monitor edges" i.e. outside visible space '<|, |>`
  - Column: `_/aaaa/b/c`, current column is in capital letters
- Examples:
  -  Simple example:
     - `<|aaB|c__|>` : Three columns: `66%, 33%, 33%` fitting in the two monitors, B active
     - Shift Window Right
     - `<|aab|C__|>` C gets highlighted columns dont move
  - "Monitor edge" example:
    - `<|aab|bCC|>d`
    - Shift window left
    - `a<|aBB|ccd|>`
      - Left selects the left window and if the wiindow is spread across two monitors tries to fit it in the left of the two monitors
      - The current column is fully on one monitor
      - The column that we moved is also fully on one monitor and visible
      
Similar behaviour can maybe be implemented for scrollable workspaces across vertical stacks of monitors?

**Comment by YaLTeR** [comment]
There's no way to combine monitors in niri right now and it would be a quite a bit of work to implement it unfortunately. Also the monitor edge handling that you describe could end up *very* complex and edge-casy I'm afraid, I'd rather not go there.

**Reply by meghprkh** [reply]
Can this be implemented with IPC (outside niri)? I am thinking of using a small custom script and changing the bindings super+left/right to fire the custom script which fetches the JSON state via IPC/`niri msg`, does all of the logic and sends the correct commands to shift windows around to niri?

**Reply by YaLTeR** [reply]
Maybe after the IPC gets window positions.
